src/consensus/tower.zig

```diff
@@ -40,12 +40,20 @@ const ComputedBankState = struct {
 
 pub const ThresholdDecision = union(enum) {
     passed_threshold,
-    failed_threshold: struct {
-        // vote depth
-        u64,
-        // Observed stake
-        u64,
-    },
+    failed_threshold: FailedThreshold,
+
+    /// NOTE: this is a tuple in the original rust code
```

Comment:
Is this comment necessary? I think it would get a bit cluttered in sig if we noted every single struct that we organized slightly differently than agave. Is there some special implication for this struct needing parity with agave's struct?

Reply:
So this is existing code. See https://github.com/Syndica/sig/blob/4a962ccea6679388d37b68894b5d579ceaf7f8c5/src/consensus/progress_map.zig#L736-L751

It only shows here as a diff as it was moved to replace the duplicated one that used a tuple.

As to the rationale, we might have to defer to @InKryption as he added this version